extensions/wikia/Wall/WallHelper.class.php

```diff
@@ -617,42 +632,20 @@ public static function getWallTitleData( $rc = null, $row = null ) {
 			$articleTitleTxt = strip_tags( $articleTitleTxt );
 		}
 
-		$ci = $wm->getCommentsIndexEntry();
-		if ( empty( $ci ) && ( $row->page_namespace == NS_USER_WALL ) ) {
-			// change in NS_USER_WALL namespace mean that wall page was created (bugid:95249)
-			$title = Title::newFromText( $row->page_title, NS_USER_WALL );
-
-			$out = [
-				'articleTitle' => $title->getPrefixedText(),
-				'articleFullUrl' => $title->getFullUrl(),
-				'articleTitleVal' => '',
-				'articleTitleTxt' => wfMessage(  'wall-recentchanges-wall-created-title' )->text(),
-				'wallTitleTxt' => $title->getPrefixedText(),
-				'wallPageFullUrl' =>  $title->getFullUrl(),
-				'wallPageName' => $row->page_title,
-				'actionUser' => $userText,
-				'isThread' => $wm->isMain(),
-				'isNew' => $isNew
-			];
-
-		} else {
-			$title = Title::newFromText( $articleId, NS_USER_WALL_MESSAGE );
-
-			$out = [
-				'articleTitle' => $title->getPrefixedText(),
-				'articleFullUrl' => $wm->getMessagePageUrl(),
-				'articleTitleVal' => $articleTitleTxt,
-				'articleTitleTxt' => empty( $articleTitleTxt ) ? wfMessage( 'wall-recentchanges-deleted-reply-title' )->text() : $articleTitleTxt,
-				'wallTitleTxt' => $wm->getArticleTitle()->getPrefixedText(),
-				'wallPageFullUrl' => $wm->getArticleTitle()->getFullUrl(),
-				'wallPageName' => $wm->getArticleTitle()->getText(),
-				'actionUser' => $userText,
-				'isThread' => $wm->isMain(),
-				'isNew' => $isNew
-			];
-		}
-
-		wfProfileOut( __METHOD__ );
+		$title = Title::makeTitle( NS_USER_WALL_MESSAGE, $articleId );
+
+		$out = [
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$out". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)